src/components/date-time-input/date-time-input.ts

```diff
@@ -0,0 +1,654 @@
+import { html } from 'lit';
+import { property } from 'lit/decorators.js';
+import { live } from 'lit/directives/live.js';
+import {
+  DatePartDeltas,
+  DatePartInfo,
+  DateParts,
+  DatePart,
+  DateTimeUtil,
+} from './date-util';
+import { watch } from '../common/decorators';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { IgcMaskInputBaseComponent } from '../mask-input/mask-input-base';
+import { partNameMap } from '../common/util';
+
+export default class IgcDateTimeInputComponent extends IgcMaskInputBaseComponent {
+  public static readonly tagName = 'igc-date-time-input';
+
+  protected _defaultMask!: string;
+  protected _value!: Date | null;
+
+  private _inputDateParts!: DatePartInfo[];
+  private _inputFormat!: string;
+  private _datePartDeltas: DatePartDeltas = {
+    date: 1,
+    month: 1,
+    year: 1,
+    hours: 1,
+    minutes: 1,
+    seconds: 1,
+  };
+
+  @property({ attribute: 'input-format' })
+  public get inputFormat(): string {
+    return this._inputFormat || this._defaultMask;
+  }
+
+  public set inputFormat(val: string) {
+    if (val) {
+      this.setMask(val);
+      this._inputFormat = val;
+    }
+  }
+
+  @property({
+    converter: (value) => {
+      if (!value) {
+        return null;
+      }
+      return DateTimeUtil.parseIsoDate(value);
+    },
+  })
+  public get value(): Date | null {
```

Comment:
I believe we add `@blazorTwoWayBind('igcChange', 'detail')` to this similar to how it's in the input, though I don't think it'll work correctly while the detail type is still `string`